bcgov/PIMS
# PIMS-1832 Error Email

## 🎯 Summary

  
[PIMS-1832](https://citz-imb.atlassian.net/browse/PIMS-1832?atlOrigin=eyJpIjoiMTAxYWNkYjVhMjBkNDAxMmI2OTE4NDliODI5YWMxN2QiLCJwIjoiaiJ9)


## Changes
- Add nunjucks template for error email
  - We could move this to the notifications templates table. Just thought it was unlikely to be a template that admins would edit if we ever implement that feature.
- Provide some defaults for the ches service. It was failing when a bcc field wasn't included.
- Updated env template and readme

## Testing
- Create an error in the frontend that results in a rendering issue
- Add this to a component's render for easy error:
```ts
<p>{(undefined as User).AgencyId}</p>
```
- Submit an error report on the page that follows
- Monitor your email address for the result

You need to make sure your ENV is set up correctly.
```
CHES_OVERRIDE_TO="Your email here"
CHES_EMAIL_ENABLED=true
ERROR_REPORT_TO=RealPropertyDivision.Disposals@gov.bc.ca or some other email for testing.
```
If you don't include the override, make sure you have NOT set the SEND_TO_LIVE env, or else you will the the real email.


## 🔰 Checklist


> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.
